chain/chain/src/types.rs

```diff
@@ -127,6 +132,122 @@ impl BlockHeaderInfo {
     }
 }
 
+/// Block economics config taken from genesis config
+pub struct BlockEconomicsConfig {
+    gas_price_adjustment_rate: Rational,
+    min_gas_price: Balance,
+    max_gas_price: Balance,
+    chain_id: String,
+}
+
+impl BlockEconomicsConfig {
+    /// Compute min gas price according to protocol version and chain id. We only upgrade gas price
+    /// for betanet, testnet and mainnet.
+    pub fn min_gas_price(&self, protocol_version: ProtocolVersion) -> Balance {
+        match self.chain_id.as_str() {
+            "betanet" | "testnet" | "mainnet" => {
+                if protocol_version < MIN_PROTOCOL_VERSION_NEP_92 {
+                    self.min_gas_price
+                } else {
+                    MIN_GAS_PRICE_NEP_92
+                }
+            }
+            _ => self.min_gas_price,
```

Comment:
Why do we need this string comparison? What networks do we not want to be affected by the change?
This looks very ugly, and unless there's absolutely no way to avoid it, we should not do it this way (i.e have the same logic for all chain ids.).

Reply:
The reason for this is that I don't want to break any test. A test usually uses some randomly generated chain id and you may want to set a custom min gas price for it to test some specific scenario. In that case you don't want the gas price to suddenly change in the test. Please enlighten me if you see a better way to do this.

Reply:
Have `self.min_gas_price` to actually be the min gas price of *the current protocol version*, and use hard coded constants for past versions instead?

Reply:
Proper alternative:
We should only return the hardcoded constant min_gas_price if the genesis protocol vesrion is less than the version in whcih the new gas price was introduced